client-ts/signalr/src/JsonHubProtocol.ts

```diff
@@ -31,4 +57,42 @@ export class JsonHubProtocol implements IHubProtocol {
     public writeMessage(message: HubMessage): string {
         return TextMessageFormat.write(JSON.stringify(message));
     }
+
+    private isInvocationMessage(message: InvocationMessage): void {
+        this.assertNotEmptyString(message.target, "Invalid payload for Invocation message.");
+
+        if (message.invocationId) {
```

Comment:
An empty string is falsey, so you'll skip `assertNotEmptyString`. Use `message.invocationId === undefined` because we're explicitly allowing only `undefined` or a **non-empty** string as the only two options.